<?xml version="1.0" ?>

<launch> 
    <arg name="rviz"               default="false" />
    <arg name="description_name"   default="bipedal_robot_description"/>

    <!-- The task file for the mpc. -->
    <arg name="taskFile"          default="$(find legged_hunter_config)/config/task/task.info"/>
    <!-- The reference related config file of the robot -->
    <arg name="referenceFile"     default="$(find legged_hunter_config)/config/command/reference.info"/>
    <!-- The URDF model of the robot -->
    <arg name="urdfFile"          default="$(find legged_hunter_description)/urdf/hunter.urdf"/>
    <!-- The file defining gait definition -->
    <arg name="gaitCommandFile"   default="$(find legged_hunter_config)/config/command/gait.info"/>

    <!-- make the files into global parameters -->
    <param name="taskFile"          value="$(arg taskFile)" />
    <param name="referenceFile"     value="$(arg referenceFile)" />
    <param name="urdfFile"          value="$(arg urdfFile)" />
    <param name="gaitCommandFile"   value="$(arg gaitCommandFile)"/>

    <!-- rviz -->
    <group if="$(arg rviz)">
        <param name="$(arg description_name)" textfile="$(arg urdfFile)"/>
        <arg name="rvizconfig" default="$(find ocs2_bipedal_robot_ros)/rviz/bipedal_robot.rviz" />
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizconfig)" output="screen" />
      </group>

    <!-- load ros controller parameters -->
    <rosparam file="$(find bipedal_controllers)/config/controllers.yaml" command="load"/>

    <!-- controller manager-->
    <node name="controller_loader" pkg="controller_manager" type="controller_manager"
        output="screen" args="load
        controllers/joint_state_controller
        controllers/joint_pos_controller"/>

    <!-- target trajectories publisher -->
    <node name="bipedal_robot_target" pkg="bipedal_controllers" type="bipedal_target_trajectories_publisher" output="screen"/>

    <!-- gait command publisher -->
    <node pkg="ocs2_bipedal_robot_ros" type="bipedal_robot_gait_command" name="bipedal_robot_gait_command"
          output="screen"/>

</launch>